docs/docs/creating-a-source-plugin.md

```diff
@@ -18,147 +18,426 @@ At a high-level, a source plugin:
+exports.createSchemaCustomization = ({ actions }) => {
+  const { createTypes } = actions
+  createTypes(`
+    type Post implements Node {
+      id: ID!
+      # create a relationship between Post and the File nodes for optimized images
+      author: Author @link(from: "author.name" by: "name") // highlight-line
+      # ... other fields
+    }
+    
+    type Author implements Node {
+      name: String!
+      post: Post @link // highlight-line
+    }`)
+}
+```
+
+You can read more about connecting foreign key fields with schema customization in the guide on [Customizing the GraphQL Schema](/docs/schema-customization/#foreign-key-fields).
+
+#### Option 2: transformation relationships
+
+Transformation relationships are used when a node is _completely_ derived from another node. An example that is common in source plugins is for transforming File nodes from remote sources, like for images from remote sources. You can read about this use case in the section below on [sourcing images from remote locations](/docs/creating-a-source-plugin/#sourcing-images-from-remote-locations).
+
+You can find more information about transformation relationships in the [Creating a Transformer Plugin guide](/docs/creating-a-transformer-plugin/#creating-the-transformer-relationship).
+
 #### Union types
 
 When creating fields linking to an array of nodes, if the array of IDs are all of the same type, the relationship field that is created will be of this type. If the linked nodes are of different types; the field will turn into a union type of all types that are linked. See the [GraphQL documentation on how to query union types](https://graphql.org/learn/schema/#union-types).
 
-#### Further specification
+### Working with data received from remote sources
+
+#### Setting media and MIME types
+
+Each node created by the filesystem source plugin includes the raw content of the file and its _media type_.
+
+[A **media type**](https://en.wikipedia.org/wiki/Media_type) (also **MIME type** and **content type**) is an official way to identify the format of files/content that is transmitted on the internet, e.g. over HTTP or through email. You might be familiar with other media types such as `application/javascript`, `application/pdf`, `audio/mpeg`, `text/html`, `text/plain`, `image/jpeg`, etc.
+
+Each source plugin is responsible for setting the media type for the nodes they create. This way, source and transformer plugins can work together easily.
+
+This is not a required field -- if it's not provided, Gatsby will [infer](/docs/glossary#inference) the type from data that is sent -- but it's the way for source plugins to indicate to transformers that there is "raw" data that can still be further processed. It also allows plugins to remain small and focused. Source plugins don't have to have opinions on how to transform their data: they can set the `mediaType` and push that responsibility to transformer plugins, instead.
```

Comment:
```suggestion
This is not a required field -- if it's not provided, Gatsby will [infer](/docs/glossary#inference) the type from data that is sent -- but it's how source plugins indicate to transformers that there is "raw" data the transformer can further process. It also allows plugins to remain small and focused. Source plugins don't have to have opinions on how to transform their data: they can set the `mediaType` and push that responsibility to transformer plugins instead.
```

Reply:
We might want to take the last two sentences and move them to the following paragraph. The transition is a bit abrupt.